astroquery/mocserver/core.py

```diff
@@ -1,416 +1,707 @@
+            If ``return_moc`` is set to True, this fixes the maximum precision order
+            of the returned MOC. For one dimensional MOCs (Space-MOCs and Time-MOCs),
+            the order is given as an integer. For Space-Time MOCs, the order should be a
+            string with the space and time orders.
+            Example: 's3 t45'
+        get_query_payload : bool, optional
+            If True, returns a dictionary of the query payload instead of the response.
+        verbose : bool, optional
+        cache: bool, optional
+            Whether the response should be cached.
 
         Returns
         -------
         response : `~requests.Response`:
-            The HTTP response from the `CDS MOCServer <http://alasky.unistra.fr/MocServer/query>`_.
+            The HTTP response from the
+            `CDS MOCServer <http://alasky.unistra.fr/MocServer/query>`_.
+
         """
-        request_payload = self._args_to_payload(**kwargs)
-        if get_query_payload:
-            return request_payload
+        request_payload = _args_to_payload(
+            meta_data=meta_data,
+            return_moc=return_moc,
+            max_norder=max_norder,
+            fields=fields,
+            max_rec=max_rec,
+            region=region,
+            intersect=intersect,
+            spacesys=spacesys,
+            casesensitive=casesensitive,
+            default_fields=self.DEFAULT_FIELDS,
+        )
 
         params_d = {
-            'method': 'GET',
-            'url': self.URL,
-            'timeout': self.TIMEOUT,
-            'data': kwargs.get('data', None),
-            'cache': False,
-            'params': request_payload,
+            "method": "GET",
+            "url": self.URL,
+            "timeout": self.TIMEOUT,
+            "cache": cache,
+            "params": request_payload,
         }
 
-        if not self.path_moc_file:
-            response = self._request(**params_d)
-        else:
-            # The user ask for querying on a MOC region.
-            with open(self.path_moc_file, 'rb') as f:
-                params_d.update({'files': {'moc': f.read()}})
-                response = self._request(**params_d)
+        if region:
+            if get_query_payload:
+                return request_payload
+            if not isinstance(region, (MOC, TimeMOC, STMOC)):
+                return self._request(**params_d)
+            # MOCs have to be sent in a multipart request
+            with NamedTemporaryFile() as tmp_file:
+                region.save(tmp_file.name, overwrite=True, format="fits")
 
-        return response
+                params_d.update({"files": {"moc": tmp_file.read()}})
+                return self._request(**params_d)
 
-    def _args_to_payload(self, **kwargs):
-        """
-        Convert the keyword arguments to a payload.
+        if get_query_payload:
+            return request_payload
+        return self._request(**params_d)
+
+    def list_fields(self, keyword=None, cache=True):
```

Comment:
```suggestion
    def list_fields(self, *, keyword=None, cache=True):
```

Reply:
I'd like to keep `keyword` as a positional argument, if you're ok with it (will be committed like this soon)